apps/cowswap-frontend/src/modules/injectedWidget/hooks/useAppCodeWithWidgetMetadata.ts

```diff
@@ -0,0 +1,46 @@
+import { useMemo } from 'react'
+
+import { environmentName } from '@cowprotocol/common-utils'
+
+import { AppDataWidget } from 'modules/appData/types'
+
+import { useInjectedWidgetMetaData } from './useInjectedWidgetMetaData'
+
+export interface AppCodeWithWidgetMetadata {
+  appCode: string
+  environment?: string
+  widget?: AppDataWidget
+}
+
+/**
+ *
+ * @param appCode of the app
+ *
+ * @returns the appCode that should be used in the metadata, and the widget metadata if it exists
+ */
+export function useAppCodeWithWidgetMetadata(appCode: string | null): AppCodeWithWidgetMetadata | null {
+  const injectedWidgetMetadata = useInjectedWidgetMetaData()
+
+  const appCodeWidget = injectedWidgetMetadata?.appCode
+
+  return useMemo(() => {
+    if (!appCode) {
+      return null
+    }
```

Comment:
Why return null for widget and env when appCode is not set?

Reply:
its mandatory field. It was already like this, i kept the same logic